cupy/core/_routines_statistics.pyx

```diff
@@ -234,49 +234,43 @@ cdef _nanargmax = create_reduction_func(
 cdef ndarray _var(
         ndarray a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
 
-    if a.dtype.kind == 'c':
-        if out is not None:
-            raise NotImplementedError(
-                'Variance for complex numbers is not implemented when out != '
-                'None. Current implemention does not convert the dtype.'
-            )
-        dtype_out = dtype
-        if dtype in ['D', 'F']:
-            dtype == dtype.lower()
-        out = _var(a.real, axis, dtype, None, ddof, keepdims)
-        out += _var(a.imag, axis, dtype, None, ddof, keepdims)
-        if dtype_out is not None and dtype_out != dtype:
-            out = out.astype(dtype_out, copy=False)
-        return out
-
     if axis is None:
         axis = tuple(range(a.ndim))
     if not isinstance(axis, tuple):
         axis = (axis,)
 
+    dtype_mean = a.dtype.char
     dtype_out = dtype
-    if dtype in ['D', 'F']:
-        if a.dtype.kind == 'f':
-            dtype = a.dtype.char
+    if dtype is None:
+        if a.dtype.kind in 'biu':
+            dtype_mean = 'd'
+            dtype_out = 'd'
```

Comment:
You should not use dtype char code.

https://docs.scipy.org/doc/numpy/reference/arrays.scalars.html#built-in-scalar-types